src/lib.rs

```diff
@@ -1,6 +1,189 @@
+    /// [is](struct.Var.html#method.is) or [isnt](struct.Var.html#method.isnt)
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// let var: environment::Var = ("FOO", "BAR").into();
+    ///
+    /// assert_eq!(var.clone().is("BAR").clear().var("FOO"), ("FOO", "").into());
+    ///
+    /// assert_eq!(var.clone().isnt("BAR").clear().var("FOO"), var);
+    ///
+    /// ```
+    pub fn clear(mut self) -> Environment {
+        if self.condition {
+            self.value.clear();
+        }
+
+        self.environment.vars.push((self.key, self.value));
+
+        self.environment
+    }
+
+    /// Append a value to the current variable's value.
+    ///
+    /// Note: appending can be conditioned by
+    /// [is](struct.Var.html#method.is) or [isnt](struct.Var.html#method.isnt)
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// let var: environment::Var = ("FOO", "BAR").into();
+    ///
+    /// assert_eq!(var.clone().is("BAR").append("BAZ").var("FOO"), ("FOO", "BARBAZ").into());
+    ///
+    /// assert_eq!(var.clone().isnt("BAR").append("BAZ").var("FOO"), var);
+    ///
+    /// ```
+    pub fn append<T: ToString>(mut self, value: T) -> Environment {
+        if self.condition {
+            self.value.push(value.to_string());
+        }
+
+        self.environment.vars.push((self.key, self.value));
+
+        self.environment
+    }
+
+    /// Define the variable's value equal to the system variable.
+    ///
+    /// Note: inheriting can be conditioned by
+    /// [is](struct.Var.html#method.is) or [isnt](struct.Var.html#method.isnt)
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// ::std::env::set_var("FOO", "BAR");
+    ///
+    /// let var: environment::Var = ("FOO", "BAZ").into();
+    ///
+    /// assert_eq!(var.clone().is("BAZ").inherit().var("FOO"), ("FOO", "BAR").into());
+    /// assert_eq!(var.clone().isnt("BAZ").inherit().var("FOO"), ("FOO", "BAZ").into());
+    ///
+    /// ```
+    pub fn inherit(mut self) -> Environment {
+        if self.condition {
+            self.value = if let Some(v) = ::std::env::vars_os()
+                .filter(|&(ref k, _)| k == &self.key)
+                .collect::<Vec<(OsString, OsString)>>()
+                .first()
+            {
+                v.1.clone()
+            } else {
+                OsString::new()
+            };
+        }
+
+        self.environment.vars.push((self.key, self.value));
+
+        self.environment
+    }
+}
+
+impl<T: ToString, Z: ToString> From<(T, Z)> for Var {
+    fn from(v: (T, Z)) -> Self {
+        Self {
+            key: OsString::from(v.0.to_string()),
+            value: OsString::from(v.1.to_string()),
+            condition: true,
+            environment: Environment::empty(),
+        }
+    }
+}
+
+impl PartialEq for Var {
+    fn eq(&self, other: &Self) -> bool {
```

Comment:
Might be kind of slick to also define `PartialEq` for tuples and individual strings.

```rust
env.var("key") == ("key", "value");
env.var("key") == "value";
```

Thoughts?

Reply:
In thinking about this more, it feels a little too magical.  Maybe instead a `.get()`.  The main benefit would be for tests so they don't have to `== (key, value).into()` but instead `.get() == value`.  Besides being briefer, I feel like it makes the intent stand out more.